src/bridging/providers/across/util.ts

```diff
@@ -29,43 +37,117 @@ export function getTokenAddress(tokenSymbol: string, chainConfig: AcrossChainCon
   return chainConfig.tokens[tokenSymbol]
 }
 
-export function toBridgeQuoteResult(amount: string, suggestedFees: SuggestedFeesResponse): AcrossQuoteResult {
-  // TODO: Do we need to use all the other fees, or they are included in totalRelayFee? I know 'lpFee' fee is, as stated in the docs, but not sure about the others.
+export function toBridgeQuoteResult(
+  request: QuoteBridgeRequest,
+  slippageBps: number,
+  suggestedFees: SuggestedFeesResponse
+): AcrossQuoteResult {
+  const { kind } = request
+
+  return {
+    isSell: kind === OrderKind.SELL,
+    amountsAndCosts: toAmountsAndCosts(request, slippageBps, suggestedFees),
+    quoteTimestamp: Number(suggestedFees.timestamp),
+    expectedFillTimeSeconds: Number(suggestedFees.expectedFillTimeSec),
+    suggestedFees,
+  }
+}
+
+function toAmountsAndCosts(
+  request: QuoteBridgeRequest,
+  slippageBps: number,
+  suggestedFees: SuggestedFeesResponse
+): BridgeQuoteAmountsAndCosts {
+  const { amount, sellTokenDecimals, buyTokenDecimals } = request
+
+  // Get the amounts before fees
+  const sellAmountBeforeFeeBig = getBigNumber(amount, sellTokenDecimals)
+  const sellAmountBeforeFee = sellAmountBeforeFeeBig.big
+  const buyAmountBeforeFee = getBigNumber(sellAmountBeforeFeeBig.num, buyTokenDecimals).big // Sell and buy token should be the same asset for current implementation, but technically they can have different decimals
+
+  // Apply the fee to the buy amount (sell amount doesn't change)
+  const totalRelayerFeePct = BigInt(suggestedFees.totalRelayFee.pct)
+  const buyAmountAfterFee = applyPctFee(buyAmountBeforeFee, totalRelayerFeePct)
+
+  // Calculate the fee
+  const feeSellToken = sellAmountBeforeFee - applyPctFee(sellAmountBeforeFee, totalRelayerFeePct)
+  const feeBuyToken = buyAmountBeforeFee - buyAmountAfterFee
+  // TODO: Do we need to use any of the other fees, or they are included in totalRelayFee? I know 'lpFee' fee is, as stated in the docs, but not sure about the others.
   // const relayerCapitalFee = suggestedFees.relayerCapitalFee
   // const relayerGasFee = suggestedFees.relayerGasFee
   // const lpFee = suggestedFees.lpFee
 
+  // Apply slippage
+  const buyAmountAfterSlippage = applyBps(buyAmountAfterFee, slippageBps)
+
   return {
-    suggestedFees,
-    buyAmount: applyFee(amount, suggestedFees.totalRelayFee.pct),
-    feeBps: pctToBps(suggestedFees.totalRelayFee.pct),
-    slippageBps: 0, // TODO: Can I set slippage to zero? are quotes enforced?
+    beforeFee: {
+      sellAmount: sellAmountBeforeFee,
+      buyAmount: buyAmountBeforeFee, // Assuming the price is 1:1 (before fee). This is because we are exchanges the same asset
```

Comment:
Typo:

```suggestion
      buyAmount: buyAmountBeforeFee, // Assuming the price is 1:1 (before fee). This is because we are exchanging the same asset
```